src/backend/joanie/core/models/contracts.py

```diff
@@ -0,0 +1,232 @@
+
+    def __str__(self):
+        return self.title
+
+    def get_body_in_html(self):
+        """Return the body in html format."""
+        if not self.body:
+            return ""
+        return markdown.markdown(textwrap.dedent(self.body))
+
+    def get_document_context(self, order):
+        """
+        Build the contract document context for the given language.
+        """
+        organization = order.organization
+        owner = order.owner
+        product = order.product
+
+        return {
+            "contract": {
+                "body": self.get_body_in_html(),
+                "title": self.title,
+            },
+            "course": {
+                "name": product.safe_translation_getter(
+                    "title", language_code=self.language
+                ),
+            },
+            "student": {
+                "name": owner.get_full_name() or owner.username,
+                "address": owner.addresses.filter(is_main=True).first(),
+            },
+            "organization": {
+                "logo": image_to_base64(organization.logo),
+                "name": organization.safe_translation_getter(
+                    "title", language_code=self.language
+                ),
+                "signature": image_to_base64(organization.signature),
+            },
+        }
+
+    def generate_document(self, order):
+        """
+        Generate the contract definition.
+        """
+        context = self.get_document_context(order)
+        file_bytes = generate_document(
+            path=enums.ISSUER_TEMPLATE_CONTRACT_DEFINITION, context=context
+        )
+        return context, file_bytes
+
+
+class Contract(BaseModel):
+    """
+    Contract represents and records the user contract issued as part of an order.
+    """
+
+    definition = models.ForeignKey(
+        to=ContractDefinition,
+        verbose_name=_("Contract definition"),
+        related_name="contracts",
+        on_delete=models.RESTRICT,
+        editable=False,
+    )
+    order = models.OneToOneField(
+        Order,
+        verbose_name=_("order"),
+        on_delete=models.PROTECT,
+        editable=False,
+    )
+
+    # Set on contract generation
+    definition_checksum = models.CharField(
+        max_length=255, editable=False, blank=True, null=True
+    )
+    context = models.JSONField(
+        _("context"),
+        blank=True,
+        help_text=_("Localized data snapshot on contract signature"),
+        editable=False,
+        null=True,
+    )
+    signature_backend_reference = models.CharField(
+        blank=True,
+        editable=False,
+        max_length=255,
+        null=True,
+        verbose_name=_("Reference in the external signature backend"),
+    )
+
+    # Set on student signature
+    signed_on = models.DateTimeField(
+        _("Date and time of issuance"), null=True, blank=True, editable=False
+    )
+
+    submitted_for_signature_at = models.DateTimeField(
```

Comment:
for coherence with "signed_on":
```suggestion
    submitted_for_signature_on = models.DateTimeField(
```